frontend/src/features/documents/documentUpload/DocumentUploadView.tsx

```diff
@@ -0,0 +1,206 @@
+import { Button } from 'components/common/buttons/Button';
+import { Input, Select, SelectOption } from 'components/common/form';
+import { Scrollable } from 'components/common/Scrollable/Scrollable';
+import TooltipIcon from 'components/common/TooltipIcon';
+import LoadingBackdrop from 'components/maps/leaflet/LoadingBackdrop/LoadingBackdrop';
+import * as API from 'constants/API';
+import { SectionField } from 'features/mapSideBar/tabs/SectionField';
+import { Formik } from 'formik';
+import useLookupCodeHelpers from 'hooks/useLookupCodeHelpers';
+import { Api_DocumentType, Api_UploadRequest } from 'models/api/Document';
+import { Api_Storage_DocumentTypeMetadataType } from 'models/api/DocumentStorage';
+import { ChangeEvent, useState } from 'react';
+import { Col, Row } from 'react-bootstrap';
+import styled from 'styled-components';
+
+interface DocumentUploadForm {
+  documentTypeId: string;
+  documentStatusCode: string;
+}
+
+interface IDocumentUploadViewProps {
+  documentTypes: Api_DocumentType[];
+  isLoading: boolean;
+  mayanMetadata: Api_Storage_DocumentTypeMetadataType[];
+  onDocumentTypeChange: (param: any) => void;
+  onUploadDocument: (uploadRequest: Api_UploadRequest) => void;
+  onCancel: () => void;
+}
+
+/**
+ * Component that provides functionality to see document information. Can be embedded as a widget.
+ */
+const DocumentUploadView: React.FunctionComponent<IDocumentUploadViewProps> = props => {
+  const [selectedFile, setSelectedFile] = useState<File | null>(null);
+
+  const documentTypes = props.documentTypes.map<SelectOption>(x => {
+    return { label: x.documentType || '', value: x.id?.toString() || '' };
+  });
+
+  const { getOptionsByType } = useLookupCodeHelpers();
+  const documentStatusTypes = getOptionsByType(API.DOCUMENT_STATUS_TYPES);
+
+  const initialFormState: DocumentUploadForm = {
+    documentTypeId: documentTypes.length > 0 ? documentTypes[0].value.toString() || '' : '',
+    documentStatusCode:
+      documentStatusTypes.length > 0 ? documentStatusTypes[0].value.toString() || '' : '',
+  };
+
+  const handleFileInput = (changeEvent: ChangeEvent<HTMLInputElement>) => {
+    // handle validations
+    if (changeEvent.target !== null) {
+      var target = changeEvent.target;
+      if (target.files !== null) {
```

Comment:
nit: there's a chance that the files array is empty (and thus not null) but `files[0]` would return undefined